apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4SamhsaMatcher.java

```diff
@@ -522,43 +524,79 @@ private static String normalizeIcd10Code(String icd10DiagnosisCode) {
   /**
    * @param procedureConcept the procedure {@link CodeableConcept} to check
    * @return <code>true</code> if the specified procedure {@link CodeableConcept} contains any
-   *     {@link Coding}s that match any of the {@link #cptCodes}, <code>false</code> if they all do
-   *     not
+   *     {@link Coding}s that match any of the {@link #cptCodes} or has unknown coding systems,
+   *     <code>false</code> otherwise
    */
   private boolean containsSamhsaProcedureCode(CodeableConcept procedureConcept) {
-    for (Coding procedureCoding : procedureConcept.getCoding()) {
-      if (TransformerConstants.CODING_SYSTEM_HCPCS.equals(procedureCoding.getSystem())) {
-        if (isSamhsaCptCode(procedureCoding)) return true;
-      } else {
-        /*
-         * Fail safe: if we don't know the procedure Coding system, assume the code is
-         * SAMHSA.
-         */
-        return true;
-      }
+    // If there are no procedure codes, then we cannot have any blacklisted codes
+    if (procedureConcept.getCoding().isEmpty()) {
+      return false;
     }
 
-    // No blacklisted procedure Codings found: this procedure isn't SAMHSA-related.
-    return false;
+    // Check that Coding to see if it's blacklisted.
+    if (hasHspcAndSamhsaCptCode(procedureConcept)) {
+      return true;
+    } else if (!containsOnlyKnownSystems(procedureConcept)) {
+      /*
+       * Fail safe: if we don't know the procedure Coding system, assume the code is
+       * SAMHSA.
+       */
+      return true;
+    } else {
+      // Otherwise, this only has known & non-SAMHSA-blacklisted procedure codes.
+      return false;
+    }
```

Comment:
Actually I just noticed, I think this could also be simplified to:
```java
if (hasHspcAndSamhsaCptCode(procedureConcept)) {
  return true;
} else {
  return !containsOnlyKnownSystems(procedureConcept));
}

Reply:
Perhaps that could be done as a documentation comment of the method just to keep the code a little more terse?  Just a thought.
```java
/**
 * This method will return true (contains SAMHSA) if the concept contains a HCPCS samhsa coding
 * or a coding with an unknown system.
 */

Reply:
Actually this made me notice this method was missing a description entirely, I'll simplify and move the explanation to the method description. Good call!